### PR TITLE
Improve netns-isolation and tests

### PR DESCRIPTION
**Pull request by erikarvstedt**
This PR is currently applied on top of [Fix process info restriction](https://github.com/fort-nix/nix-bitcoin/pull/225), so that the tests don't fail locally.

The only functional change in this PR is that it's now possible to test all scenarios at once via `run-test.sh`.
Also, the PR fixes a few issues that prevent modules-only usage of nix-bitcoin.

### CODE REVIEW

File: modules/netns-isolation.nix [hunk 1]
```diff
@@ -68,117 +75,88 @@ in {
...
+
+    allowedUser = mkOption {
+      type = types.str;
+      description = ''
+        User that is allowed to execute commands in the service network namespaces.
+        The user's group is also authorized.
+      '';
+    };
+
+    netns = mkOption {
```
Comment (nixbitcoin) [inline modules/netns-isolation.nix:87]: Why is this needed?

Comment (erikarvstedt) [reply modules/netns-isolation.nix:87]: You mean the `netns` option?
See [netns: move webindex config for modules-only usage](https://github.com/fort-nix/nix-bitcoin/pull/226/commits/e1d60fdeea3fcc5e9bd5985fdd17db7d8912652f). It's used by `nix-bitcoin-webindex` to access the nginx netns address.

Comment (nixbitcoin) [reply modules/netns-isolation.nix:87]: Ah ok, so this exposes network namespace addresses to all modules? Very useful.

**Comment by nixbitcoin** [comment]
Sorry for breaking modules-only usage in so many places. I simply didn't have it in mind because I had never used nix-bitcoin in that way. Maybe we should document nix-bitcoins architecture to also communicate this use-case to new developers.

Would it be possible to add a modules-only test scenario where we can test that new changes don't break this use-case?

**Comment by erikarvstedt** [comment]
Yes, @jonasnick also proposed this [here](https://github.com/fort-nix/nix-bitcoin/pull/225#pullrequestreview-471989636). I'll add a test and an example.

File: modules/netns-isolation.nix [hunk 2]
```diff
@@ -241,66 +265,40 @@ in {
...
-      # Switch user because lnd makes datadir contents readable by user only
-      ''
-        netns-exec nb-lnd sudo -u lnd ${config.services.lnd.package}/bin/lncli --tlscertpath ${config.nix-bitcoin.secretsDir}/lnd-cert \
-          --macaroonpath '${config.services.lnd.dataDir}/chain/bitcoin/mainnet/admin.macaroon' "$@"
-      '';
+      cliExec = mkCliExec "lnd";
     };
 
-    # liquidd: Custom netns configs
-    services.liquidd = mkIf config.services.liquidd.enable {
```
Comment (nixbitcoin) [inline modules/netns-isolation.nix:253]: Hey, I'm afraid these `mkIf`s are necessary since many of the options only exist if the modules are enabled. Otherwise you get an `error: attribute 'lnd' missing` for example.

File: modules/netns-isolation.nix [hunk 3]
```diff
@@ -53,12 +53,11 @@ in {
       type = types.attrsOf (types.submodule {
         options = {
           id = mkOption {
-            # TODO: Exclude 10
             # TODO: Assert uniqueness
-            type = types.int;
+            type = types.ints.between 10 255;
```
Comment (nixbitcoin) [inline modules/netns-isolation.nix:65]: This includes 10 and 255, shouldn't it be `types.ints.between 11 255;`?

Comment (erikarvstedt) [reply modules/netns-isolation.nix:65]: Hah, that's what reviews are for :cat:. Thanks for catching this.

File: modules/netns-isolation.nix (hunk 2 above)
Comment (erikarvstedt) [reply modules/netns-isolation.nix:253]: It works, because, thanks to nix' lazyness, the option values are never evaluated for disabled modules.

Comment (nixbitcoin) [reply modules/netns-isolation.nix:253]: Hmm, I'm getting `error: attribute 'lnd' missing` when enabling only clightning. How can that be?

Comment (erikarvstedt) [reply modules/netns-isolation.nix:253]: I can't reproduce this.
Here's a script that successfully evals a secure-node config with only clightning enabled. Run it from within nix-shell.
```bash
read -d '' tmpstr <<'EOF'
(import <nixpkgs/nixos> {
configuration = { config, lib, ... }: with lib; {
    imports = [
      <nix-bitcoin/modules/presets/secure-node.nix>
    ];

    nix-bitcoin.netns-isolation.enable = true;

    services.clightning.enable = true;
    # services.spark-wallet.enable = true;
    # services.lightning-charge.enable = true;
    # services.nanopos.enable = true;
    # services.lnd.enable = true;
    # services.lnd.listenPort = 9736;
    # services.lightning-loop.enable = true;
    # systemd.services.lightning-loop.wantedBy = mkForce [];
    # services.electrs.enable = true;
   # services.backups.enable = true;
  };
}).vm.outPath
EOF
nix eval "($tmpstr)"
```

Comment (erikarvstedt) [reply modules/netns-isolation.nix:253]: You're probably referencing `services.lnd.listen` somewhere in your config, even when `lnd` is disabled.

Comment (nixbitcoin) [reply modules/netns-isolation.nix:253]: Hmm, your script does run fine, however I'm getting

```
while evaluating 'mkCliExec' at /nix-bitcoin/modules/netns-isolation.nix:46:15, called from /nix-bitcoin/modules/n:
attribute 'lnd' missing, at /nix-bitcoin/modules/netns-isolation.nix:46:43
```

when deploying with `nixops deploy`

Comment (erikarvstedt) [reply modules/netns-isolation.nix:253]: Fixed. The bug was in webindex.

**Comment by nixbitcoin** [comment]
Ok, I think we can squash the fixups and I'll ACK.

**Comment by erikarvstedt** [comment]
Squashed and rebased.

**Review by nixbitcoin — APPROVED** [review]
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 09ec03ca4b384f50daf6f005dc296ae9b337539e
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEEV3o0Un8+KoXoD+Fk3RH5rVMIs7oFAl9E384ACgkQ3RH5rVMI
s7oB+w//TQen7NBEBiKIhilrbD7MNRm24LHyVb787DOx3hSQy3bBpYUphxrknemh
gEyGcjYIGREnB/erRebIGnlWzS+TXjjPnIG589Tl591LCjZKGJs07EIVx2h+a4vD
wCX2zbbwedqwE+cH3ZiLnk0RTtCDV907XIsNAK2CdlGImksL9RDh/KQzPQg3dKP+
X9E4gGyUbFaA3TqHPRmorghi/Kw//+m22Xw7Lcfdo4DFzY51wI6d7RW7y9vi3dyM
YPjj5VAh+oXkR93zR1Dz9FqTJ89ZiS2mOJ+6hJk0oXAmxSy5jVdRsGUpxGJJoGa7
dglEofhGWt18FzQUNo+uyspRVX9vpa0KicbV+VvHwkhqMPAV9ytHzoNM1dPLy3H7
iJyQVdZzv7Dhk8ctfLMW2to3PLU58NuCYbCX7Ga86HO2eaP5wwUoNzO7rOFMmvQv
SlzXt0r7MO3oG8HVzNPsAGI9noW9A3k0vc2U1ocyRRlrYrWJp21Uo2YFmrZ/dASO
c5AbEdvhN0gNFg2GqFTDiP9nSWRP4OrOZdU8eH4hOF6Thz8AVhI4f4XepzcK32mH
gFMAYRIpNKwhuHkeBcQlzJZIdKfYChzt7CXKBYUueMhCrvmyl9SCuOP18W6f7H8b
J18i0RIO+mzXytlc9a3Kh9IIXxOKunV+qZ9Nxv+yIOv5in0FGVY=
=ZYQQ
-----END PGP SIGNATURE-----
```

**Review by jonasnick — COMMENTED** [review]
Very nice! Can you remove the tabs from run-tests? Micro-nit: br0 was renamed nb-br, not br-nb as the commit message says. Otherwise looks good.

**Comment by erikarvstedt** [comment]
@jonasnick, fixed. µ-nits are always welcome!

**Review by jonasnick — APPROVED** [review]
ACK e5fb3f6a7fb0f32ec036d2eebfc239528b0345af

**Review by nixbitcoin — APPROVED** [review]
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK e5fb3f6a7fb0f32ec036d2eebfc239528b0345af
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEEV3o0Un8+KoXoD+Fk3RH5rVMIs7oFAl9FD98ACgkQ3RH5rVMI
s7rUsg//Q1ARK+G3t1Xt+0faIb5VTN5yfhLwEOUW3lKY703e7SRhEtWrjXOUoXLd
q+YsZ7HJQ2zT5SihL4lkPBImazaPHWzLwV18+fWgfJ+ubCAQObKWk6xlSLlwrxDN
uM+j8BNQ74O++/4X4p1NYEod+lCR/2eU+Aj+hQuHskfTNFj64LXHWSYncvB8wx44
4MoG43Jk76PRJvW+Lv1VWf16M9mQOc/JGHJzgBhgDHBBxAJn3SCvBL8acG8cebhZ
l9jIyL+z+utSTaMqgmy0Ewi59PXuH6AnpiQ+LsgIDbAElzsJinqwEmIO0CI2cO7D
5sZFMdEmkDtYpgLJdePa4vzVWyV1SH15Xy+oZQZt/RQPTV6ONoZaHtHlr9sc2jLF
xS94Oad97fcmiB+7UaXMTHhUiR1/VYl7HBZVp6sGxjOBXE1VRdCPrfu9mCZh9gRL
FARUdcBqH/zJlR0eFol+uaB/apCFqew/Ot+kna3tLzZh13maaF9sCR7XwmlygfeU
lgCuwGemOeE3OkdMQI1ZY9amE/dj2QsH5uKBb6IxdoOOL899OULHTzWwTIiD37zN
EnMZtw22FJCrWOFB9nzhk6mdrTQ/QUEbFr3gSVYszN+cOuey2QJv3ZV5ZDhEtEdJ
7UgPcuUaPiaPuDEjrVXZa3yY/dIIZAUkOufNoR1ejw3V9oH/VdE=
=tUam
-----END PGP SIGNATURE-----
```
